HACKING.md

```diff
@@ -0,0 +1,36 @@
+Coding style Guidelines
+=======================
+
+Follow [PEP-8](https://www.python.org/dev/peps/pep-0008/) coding guidelines.
+
+
+Commit Message Guidelines
+=========================
+
+Commit Guidelines inspired by [Gnome Commit Guidelines](https://wiki.gnome.org/Git/CommitMessages).
+
+Those are only general-purpose recommended guidelines, depending on the context of each PR the following rules can vary. 
+
+Remember: the commit message is mainly for the other people, so they should be able to understand the changes made at any point in time.
+
+Example
+-------
+
+```
+short explanation of the commit
+
+Longer (optional) explanation explaining exactly what's changed and why, whether any
+external or private interfaces changed, what bugs were fixed (with bug
+tracker reference if applicable) and so forth. Be concise but not too brief. Avoid writing long lines, use newlines when necessary.
+
+[Reference to the issue solved, if any]
+```
+
+Details
+-------
+
+- First line (the brief description) must only be one sentence and should start with a lowercase letter unless it starts with a capital symbol. The message should be concise, try not to exceed 72 characters.
```

Comment:
I don't think either should be inforced, there are many open source projects that use both styles (see Kubernetes as an example: https://github.com/kubernetes/kubernetes/commits/master?after=cc568f6433f2d90b30ca8e1d51f3597d60affb11+34 ) I think that leaving it up to the author should be the best. I use lower case because it's faster than having to capitalize, but I believe it's more of a personal opinion and style and it shouldn't be inforced. Developers should be able to decide.